cpp/src/arrow/sparse_tensor.h

```diff
@@ -228,6 +230,73 @@ class ARROW_EXPORT SparseCSRIndex : public internal::SparseIndexBase<SparseCSRIn
   std::shared_ptr<Tensor> indices_;
 };
 
+// ----------------------------------------------------------------------
+// SparseCSFIndex class
+
+/// \brief EXPERIMENTAL: The index data for a CSF sparse tensor
+///
+/// A CSF sparse index manages the location of its non-zero values by set of
+/// prefix trees. Each path from a root to leaf forms one tensor non-zero index.
+/// CSF is implemented with three vectors.
+///
+/// Vectors inptr and indices contain N-1 and N buffers respectively, where N is the
+/// number of dimensions. Axis_order is a vector of integers of legth N. Indptr and
+/// indices describe the set of prefix trees. Trees traverse dimensions in order given by
+/// axis_order.
+class ARROW_EXPORT SparseCSFIndex : public internal::SparseIndexBase<SparseCSFIndex> {
+ public:
+  static constexpr SparseTensorFormat::type format_id = SparseTensorFormat::CSF;
+
```

Comment:
How do you think about adding `kTypeName` static constant member here likewise [this change](https://github.com/apache/arrow/pull/5932/files#diff-b49bb494ab5354bdf56064136a7355a1R291)?

Reply:
I don't mind, but I also don't see immediate use, but I trust you do :).
I added it like in ARROW-4225.